rust/arrow/src/buffer.rs

```diff
@@ -471,6 +487,9 @@ impl PartialEq for MutableBuffer {
         if self.len != other.len {
             return false;
         }
+        if self.capacity != other.capacity {
```

Comment:
Same issue .. memcmp compares up to `len` so it doesn't matter if the capacities match or not